relay-server/src/actors/project.rs

```diff
@@ -901,19 +905,23 @@ impl Project {
             _ => self.state = Some(state.clone()),
         }
 
-        // Flush all queued `ValidateEnvelope` messages
-        while let Some((envelope, context)) = self.pending_validations.pop_front() {
-            self.flush_validation(envelope, context, state.clone());
-        }
+        if state.invalid() {
+            ProjectCache::from_registry().send(RequestUpdate::new(self.project_key, no_cache));
```

Comment:
Either place the channel back into `self.state_channel`, or pop it in the `else` branch.